### ISSUE TITLE
`object.Error@(0): Access Violation` on `DMD32`

### ISSUE THREAD

**Issue by ZHANITEST** [Dgame/Dgame#76]
i was copy & paste this code from [here(tutorial/game_loop_event_handling)](http://dgame-dev.de/index.php?controller=learn&mode=tutorial&version=0.6&tutorial=game_loop_event_handling):
```
import std.stdio;

import Dgame.Window;
import Dgame.Window.Event;
import Dgame.System.Keyboard;

void main() {
    Window wnd = Window(640, 480, "Dgame Test");

    bool running = true;

    Event event;
    while (running) {
        wnd.clear();
        
        while (wnd.poll(&event)) {
            switch (event.type) {
                case Event.Type.Quit:
                    writeln("Quit Event");
                    running = false;
                break;
                    
                case Event.Type.KeyDown:
                    writeln("Pressed key ", event.keyboard.key);
                    
                    if (event.keyboard.key == Keyboard.Key.Esc)
                        running = false; // or: wnd.push(Event.Type.Quit);
                    
                break;
                    
                default: break;
            }
        }
        
        wnd.display();
    }
}
```

And, i got a another result each the D compilers...
# DMD
*  Windows 10 x64
*  "dgame": "~>0.6.6"
*  "DMD32 D Compiler v2.085.0"
```
dub run --compiler=dmd
Performing "debug" build using dmd for x86.
derelict-util 2.1.0: target for configuration "library" is up to date.
derelict-gl3 1.0.24: target for configuration "library" is up to date.
derelict-sdl2 1.9.5: target for configuration "library" is up to date.
dgame 0.6.6: target for configuration "lib" is up to date.
sample ~master: building configuration "application"...
Linking...
To force a rebuild of up-to-date targets, run again with --force.
Running .\sample.exe

object.Error@(0): Access Violation
Program exited with code 1
```
# LDC2
*  Windows 10 x64
*  "dgame": "~>0.6.6"
*  "LDC - the LLVM D compiler (1.15.0-beta1): based on DMD v2.085.0 and LLVM 8.0.0"
```
dub run --compiler=ldc2
Performing "debug" build using ldc2 for x86_64.
derelict-util 2.1.0: target for configuration "library" is up to date.
derelict-gl3 1.0.24: target for configuration "library" is up to date.
derelict-sdl2 1.9.5: target for configuration "library" is up to date.
dgame 0.6.6: target for configuration "lib" is up to date.
tongpoo ~master: building configuration "application"...
To force a rebuild of up-to-date targets, run again with --force.
Running .\tongpoo.exe
SDL version: 2.0.9
Derelict loaded GL version: 45 (45), available GL version: 4.6.13547 Compatibility Profile Context 25.20.15025.1002
Quit Event
```

i dont know why... 🤔  pls help me!
thank you

**Comment by Dgame** [comment]
That is odd. So it does work with LDC but not with DMD?

**Comment by Dgame** [comment]
It works for me on Windows 10 Pro, DMD 2.085.0 Since it seems to work in ldc I'm quite sure you haven't forgotten any libraries. So that case really is odd. Maybe you could provide me the dub.json as well?

**Comment by ZHANITEST** [comment]
> That is odd. So it does work with LDC but not with DMD?

yes.

> It works for me on Windows 10 Pro, DMD 2.085.0 Since it seems to work in ldc I'm quite sure you haven't forgotten any libraries. So that case really is odd. Maybe you could provide me the dub.json as well?

sure. i just append a line for **dependencies**.

```
{
	"description": "-",
	"license": "LGPL-v2.1",
	"authors": [
		"ZHANITEST"
	],
	"dependencies":{
		"dgame": "~>0.6.6"
	},
	"copyright": "Copyright © 2019, ZHANITEST",
	"name": "example"
}
```

**Comment by Dgame** [comment]
Well, I get "No target name set".  What worked for me is:

```json
{
    "description": "-",
    "license": "LGPL-v2.1",
    "authors": [
        "ZHANITEST"
    ],
    "configurations": [
        {
            "name": "main",
            "targetName": "dgame-example",
            "targetType": "executable",
            "sourceFiles": ["example.d"]
        }
    ],
    "dependencies":{
        "dgame": "~>0.6.6"
    },
    "copyright": "Copyright © 2019, ZHANITEST",
    "name": "example"
}
```

If that doesn't help you, maybe you should consider to post in [D learn](https://forum.dlang.org/group/learn) to get a faster response since I'm not that involved with D anymore.

**Comment by Dgame** [comment]
Here is my test folder: [dgame.zip](https://github.com/Dgame/Dgame/files/2975595/dgame.zip)

**Comment by Windspar** [comment]
This could be the audio bug. #71 
A new release could solve this.
Bug not fix in 0.6.6.

**Comment by Dgame** [comment]
>This could be the audio bug. #71
A new release could solve this.
Bug not fix in 0.6.6.

It is now in 0.6.7.
